wire-ios-data-model/Source/MLS/SafeCoreCrypto.swift

```diff
@@ -0,0 +1,53 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import CoreCryptoSwift
+
+// MARK: - Protocols
+
+public protocol SafeCoreCryptoProtocol {
+    func perform<T>(_ block: (CoreCryptoProtocol) throws -> T) rethrows -> T
+    func unsafePerform<T>(_ block: (CoreCryptoProtocol) throws -> T) rethrows -> T
+}
+
+public struct SafeCoreCrypto: SafeCoreCryptoProtocol {
+
+    private let coreCrypto: CoreCryptoProtocol
+    private let safeContext: SafeFileContext
+
+    init(coreCrypto: CoreCryptoProtocol, coreCryptoConfiguration: CoreCryptoConfiguration) {
```

Comment:
Perhaps we can have a convenience init that will create the instance of `CoreCrypto` using the configuration, what do you think?

Reply:
yep I can do it

Reply:
that would remove the `CoreCryptoFactory.createCoreCrypto` method

Reply:
Sure, or we can wrap the init of `SafeCoreCrypto` inside the factory.